test/collaboration.test.js

```diff
@@ -322,6 +336,9 @@ describe('slate operations propagate between editors', () => {
       const src = TestEditor.create();
       const dst = TestEditor.create();
 
+      src.syncDoc.set("data", new Y.Map())
+      src.syncDoc.set("document", new Y.Array())
```

Comment:
Can these two lines be folded into TestEditor.create()?

Reply:
If these changes are made inside the TestEditor.create(), we start to get weird behaviors such as running one test without any changes multiple times and getting different results at the end.

![image](https://user-images.githubusercontent.com/20113585/102272809-1b917700-3f00-11eb-9547-16b605235652.png)
![image](https://user-images.githubusercontent.com/20113585/102272885-3f54bd00-3f00-11eb-8a55-14def8e8757a.png)
![image](https://user-images.githubusercontent.com/20113585/102272958-5abfc800-3f00-11eb-80b9-768381e6eb5c.png)

I wasn't able to find out why this happens. Because of that, I kept the Doc and TopLevel Y.Map 'content' initiation inside the TestEditor.Create(), and all other changes outside to be able to propagate consistently.

Reply:
Interesting! Thanks for the extra data.

I was also wondering why you were only creating those for the src editor, not also for the dst editor.
But maybe this makes sense -- if you create them for the dst editor as well, then perhaps Yjs sees the two different instances (of both "document" and "data") of *different* things (that happen to look the same) -- versus replicated instances of the *same* thing.
Perhaps creating them only for the src editor means that Yjs takes care of replicating them to the dst editor, which means it "knows" that they are replicated instances of the "same thing".

I wonder if "document" and "data" were top-level children of the Y.doc (i.e., obtained with Y.getArray and Y.getMap, as was done for 'content' prior to this change), then we wouldn't need to populate them with explicitly created Y.{Array,Map} values -- *and*, for bonus, Yjs would simply *know* that the instances in src and dst were replicated instances of the same thing. (And if so, perhaps that would be slightly cleaner code?)

What do you think?

Reply:
About this, talking with @chrisfsampaio, he highlighted that at the server the code watches the 'content' Array, and making both 'document' and 'data' top-level child would force us to duplicate the watch code to make sure the server could see change on both ends. So he suggested creating the top-level 'content' YMap and create the other fields inside it, this way the server would only need to watch deep the top-level 'content' and catch changes on both 'data' and 'document'. 

This really depends on how these changes would impact the server's behavior. The way we made it now save us more changes on the server but adds complexity I guess.

Reply:
I don't think this is a huge deal either way, so I'm fine with the approach as is.